From 2c987545a88507acdd8a572a3bd23a4ca0124d14 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Thu, 13 Apr 2023 17:41:11 +0100
Subject: [PATCH] drm/vc4: Treat zero sized destination as full screen

Kodi video planes come through with all zeros for fullscreen
Without this check, we WARN when writing width-1, height-1
to destination dlist

Signed-off-by: Dom Cobley <popcornmix@gmail.com>
---
 drivers/gpu/drm/vc4/vc4_plane.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/gpu/drm/vc4/vc4_plane.c
+++ b/drivers/gpu/drm/vc4/vc4_plane.c
@@ -484,6 +484,11 @@ static int vc4_plane_setup_clipping_and_
 	vc4_state->crtc_w = state->dst.x2 - state->dst.x1;
 	vc4_state->crtc_h = state->dst.y2 - state->dst.y1;
 
+	if (!vc4_state->crtc_w)
+		vc4_state->crtc_w = state->crtc->mode.hdisplay;
+	if (!vc4_state->crtc_h)
+		vc4_state->crtc_h = state->crtc->mode.vdisplay;
+
 	ret = vc4_plane_margins_adj(state);
 	if (ret)
 		return ret;
